src/TypeScript/EslintBridgeClient/EslintBridgeStartUp.cs

```diff
@@ -19,17 +19,183 @@
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator;
 
 namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
 {
-    // TODO: https://github.com/SonarSource/sonarlint-visualstudio/issues/2256
-
     internal interface IEslintBridgeStartUp : IDisposable
     {
         /// <summary>
         /// Ensures that the process is running. Returns the port that server is running on.
         /// </summary>
         Task<int> Start();
     }
+
+    [Export(typeof(IEslintBridgeStartUp))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class EslintBridgeStartUp : IEslintBridgeStartUp
+    {
+        internal const string EslintBridgeDirectoryMefContractName = "SonarLint.TypeScript.EsLintBridgeServerPath";
+        
+        private readonly string eslintBridgeStartupScriptPath;
+        private readonly INodeLocator nodeLocator;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+        private TaskCompletionSource<int> startTask;
+
+        internal Process Process;
+
+        [ImportingConstructor]
+        public EslintBridgeStartUp([Import(EslintBridgeDirectoryMefContractName)] string eslintBridgeStartupScriptPath,
+            INodeLocator nodeLocator,
+            ILogger logger)
+            : this(eslintBridgeStartupScriptPath, nodeLocator, new FileSystem(), logger)
+        {
+        }
+
+        internal EslintBridgeStartUp(string eslintBridgeStartupScriptPath,
+            INodeLocator nodeLocator,
+            IFileSystem fileSystem,
+            ILogger logger)
+        {
+            this.eslintBridgeStartupScriptPath = eslintBridgeStartupScriptPath;
+            this.nodeLocator = nodeLocator;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public Task<int> Start()
+        {
+            if (startTask != null && !startTask.Task.IsFaulted)
+            {
+                return startTask.Task;
+            }
+
+            startTask = new TaskCompletionSource<int>();
+            try
+            {
+                StartServer();
+            }
+            catch (Exception ex)
+            {
+                startTask.SetException(ex);
+            }
+
+            return startTask.Task;
+        }
+
+        public void Dispose()
+        {
+            Stop();
+        }
+
+        private void StartServer()
+        {
+            var nodePath = FindNodePath();
+            var serverStartupScriptLocation = FindServerScriptFile();
+
+            var psi = new ProcessStartInfo
+            {
+                FileName = nodePath,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false, // required if we want to capture the error output
+                ErrorDialog = false,
+                CreateNoWindow = true,
+                Arguments = serverStartupScriptLocation
+            };
```

Comment:
The `ts\bin\server` script can handle multiple other arguments. Do we need to supply any of them (e.g. working directory, shouldUseTypeScriptParserForJS)?

Reply:
I'll check 👍 I assumed that was all based on the prototype, I haven't checked what else is supported.